cpp/src_prims/selection/knn.cuh

```diff
@@ -365,8 +365,8 @@ __global__ void class_probs_kernel(OutType *out, const int64_t *knn_indices,
 template <typename OutType = int>
 __global__ void class_vote_kernel(OutType *out, const float *class_proba,
                                   int *unique_labels, int n_uniq_labels,
-                                  size_t n_labels, size_t n_samples,
-                                  int n_outputs, int output_offset) {
+                                  size_t n_samples, int n_outputs,
```

Comment:
did the old version have a bug by using n_labels or was it just that `n_labels==n_samples` and they are redundant?

Reply:
`n_samples` is the number of query data samples (which is the number of rows in the given knn_indices, to make things even more confusing). n_labels is the number of index data samples, such that the labels can be looked up by scanning through the `knn_indices`. The verbiage was definitely very confusing here. I should have been using `index` and `query` / `search` from the start.